spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -981,6 +989,22 @@ def notify_close(self):
             return params
 
     # -------------------------------------------------------------------------
+    def set_debug_panel(self, debug_panel, language):
+        """Enable/disable debug panel"""
+        debugger_panel = self.panels.get(DebuggerPanel)
+        if language == 'py':
+            debugger_panel.setVisible(True)
+        else:
+            debugger_panel.setVisible(False)
+
+    def set_folding_panel(self, folding, language):
+        """Enable/disable debug panel"""
+        folding_panel = self.panels.get(FoldingPanel)
+        if language == 'py':

```

Comment:
Depending on @ccordoba12 response to the validation for language in the folding panel, please change this to:
1. If no language validation is needed something like:
```python
folding_panel.setVisible(folding)
```
2. If the language validation is needed, change the validation for something like:
```python
if language == 'py' and folding:
```